src/components/DataTypes/enum.tsx

```diff
@@ -5,74 +5,16 @@ import * as React from 'react';
 import { DataTypeEditProps, SelectProps } from 'rendition';
 import * as utils from '../../utils';
 import Select from '../Select';
-
-export const operators = {
-	is: {
-		getLabel: (_s: JSONSchema6) => 'is',
-	},
-	is_not: {
-		getLabel: (_s: JSONSchema6) => 'is not',
-	},
-};
-
-type OperatorSlug = keyof typeof operators;
-
-interface EnumFilter extends JSONSchema6 {
-	title: OperatorSlug;
-	properties?: {
-		[k: string]: {
-			const?: any;
-			not?: {
-				const?: any;
-			};
-		};
-	};
-}
-
-export const decodeFilter = (
-	filter: EnumFilter,
-): {
-	field: string;
-	operator: OperatorSlug;
-	value: any;
-} | null => {
-	const operator = filter.title;
-
-	if (!filter.properties) {
-		return null;
-	}
-
-	const keys = Object.keys(filter.properties);
-	if (!keys.length) {
-		return null;
-	}
-	let value: string;
-
-	const field = keys[0];
-
-	if (operator === 'is') {
-		value = filter.properties[field].const;
-	} else if (operator === 'is_not') {
-		value = filter.properties[field].not!.const;
-	} else {
-		return null;
-	}
-
-	return {
-		field,
-		operator,
-		value,
-	};
-};
+import { decodeFilter, OneOfFilter, operators, OperatorSlug } from './oneOf';
 
 export const createFilter = (
```

Comment:
Does this mean there is no more `EnumFilter`? I don't understand why we removed the `EnumFilter` type here and why we are exporting `oneOf` functions from the `enum` file. I think there should be a helper file that contains commonalities between the two.

Reply:
@sradevski I did it because oneOf is more generic form of enum. Enum is oneOf with same const and title. I'll probably revert changes in enum file.

Reply:
@karaxuna If that's the case, then we should somehow merge the two in a clear manner. I still see references to Enum in several places, so we either keep the two separate for this PR and merge later, or do it cleanly now, up to you :smile: